scripts/Synapse/extractTSV.ipynb

```diff
@@ -0,0 +1,336 @@
+        "    row_dict[\"recent_covid_test_1\"] = sorted_tests[0][0]\n",
+        "    row_dict[\"recent_covid_test_2\"] = sorted_tests[1][0]\n",
+        "    return row_dict\n",
+        "\n",
+        "def rewrite_test_x_with_test_y(row_dict, test_to_rewrite):\n",
+        "    \"\"\"\n",
+        "    Given a dictionary-formatted spark RDD row that's had recent covid tests identified,\n",
+        "    replaces the values of test 1 and test 2 with the information pertaining to the\n",
+        "    most recent and second most recent covid tests, respectively.\n",
+        "    \"\"\"\n",
+        "    replace_with_test = row_dict[\"recent_covid_test_\" + str(test_to_rewrite)]\n",
+        "    if replace_with_test is None:\n",
+        "        return row_dict\n",
+        "    for test_field in TEST_TEMPLATE:\n",
+        "        row_dict[test_field + str(test_to_rewrite)] = row_dict[test_field + replace_with_test]\n",
+        "    return row_dict\n",
+        "    \n",
+        "\n",
+        "# Explicitly define new schema to prevent interpretation breakage during RDD map\n",
+        "new_schema = ecr.schema.add(StructField(\"covid_tests\", StringType(), True))\n",
+        "new_schema = new_schema.add(StructField(\"recent_covid_test_1\", StringType(), True))\n",
+        "new_schema = new_schema.add(StructField(\"recent_covid_test_2\", StringType(), True))\n",
+        "\n",
+        "# Apply map functions to parallelize row processing and identify covid tests\n",
+        "test_cols = [c for c in ecr.columns if \"test_type\" in c]\n",
+        "rdd2 = ecr.rdd.map(lambda x: filter_tests(x, test_cols))\n",
+        "rdd3 = rdd2.map(lambda x: identify_recent_tests(x))\n",
+        "rdd4 = rdd3.map(lambda x: rewrite_test_x_with_test_y(x, 1))\n",
+        "rdd5 = rdd4.map(lambda x: rewrite_test_x_with_test_y(x, 2))\n",
+        "\n",
+        "# Convert back to DF and drop the temp cols we created as well as superfluous tests\n",
+        "ecr = spark.createDataFrame(rdd5, new_schema)\n",
+        "cols_to_drop = [\"covid_tests\", \"recent_covid_test_1\", \"recent_covid_test_2\"]\n",
+        "for i in range(3,21):\n",
+        "    for col_name in TEST_TEMPLATE:\n",
+        "        cols_to_drop.append(col_name + str(i))\n",
+        "ecr = ecr.drop(*cols_to_drop)\n",
+        "\n",
+        "# Now, rename all the processed columns to what LAC would like in their output\n",
+        "ecr = ecr.select(\n",
+        "    col(\"incident_id\").alias(iris_tsv_column_aliases[\"incident_id\"]),\n",
+        "#     col(\"iris_id\"),\n",
+        "    col(\"patient_id\"),\n",
+        "    col(\"person_id\"),\n",
+        "    col(\"first_name\").alias(iris_tsv_column_aliases[\"first_name\"]),\n",
+        "    col(\"last_name\").alias(iris_tsv_column_aliases[\"last_name\"]),\n",
+        "    lit(\"[ListSectionInsert]\").alias(\"Section action\"),\n",
+        "    lit(None).cast('string').alias(\"Section instance\"),\n",
```

Comment:
Curious about this and the `comments` below - is that data just not ever available?